test/test_nn.py

```diff
@@ -622,6 +622,16 @@ def hook(mod, grad_input, grad_output):
         with self.assertRaisesRegex(RuntimeError, "where no input requires gradient"):
             mod(inp).sum().backward()
 
+    def test_hook_last_arg_requires_grad(self):
+        mod = nn.L1Loss()
+        inp = torch.rand(1, requires_grad=True)
+        mod.register_full_backward_hook(lambda m, gI, gO: None)
+
+        try:
+            mod(inp.detach(), inp)
+        except Exception as ex:
+            self.fail("Unexpected exception: %s" % ex)
```

Comment:
For code that should not raise exceptions, we usually don't protect it and add a comment saying so. But this works as well.